090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,113 @@
+Any other value will raise an error and the application will fail to start with an appropriate error message.
+
+When running in standalone mode with `strimziMetricsReporter`, the user will be able to configure any Strimzi Metrics Reporter property using the `kafka.` prefix.
+At a minimum `kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter` and `kafka.prometheus.metrics.reporter.listener.enable=false` will be needed.
+The following example will be provided as a comment in the default properties file.
+
+```sh
+bridge.metrics=strimziMetricsReporter
+kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter
+kafka.prometheus.metrics.reporter.listener.enable=false
+kafka.prometheus.metrics.reporter.allowlist=.*
+```
+
+The `KAFKA_BRIDGE_METRICS_ENABLED` env var will be deprecated and removed in a future release.
+When set, the user will get a warning suggesting to use the new `bridge.metrics` configuration.
+In case they are both set, the `bridge.metrics` configuration will take precedence over `KAFKA_BRIDGE_METRICS_ENABLED`.
+
+The `MetricsReporter` class will be updated to also include a new `StrimziCollectorRegistry` that will work similarly to the `JmxCollectorRegistry`.
+The `StrimziCollectorRegistry` will include a reference to `PrometheusRegistry.defaultRegistry`, which is the same instance used by the Strimzi Metrics Reporter to collect metrics.
+
+The `kafka_bridge_config_generator.sh` script is used to generate the image configuration based on environment variables.
+This script will be updated to also include `bridge.metrics` and related configurations (see the following section).
+
+The Bridge will also try to load the JMX Exporter configuration file from the path specified by the `bridge.metrics.jmx.exporter.config.path` property.
+If not specified or not found, it will fall back to the hard coded configuration.
+
+### Cluster Operator
+
+Like for all the other major components, the `KafkaBridge` CRD will also include `metricsConfig` with support for `strimziMetricsReporter` type.
+
+Three new environment variables will be introduced to pass the metrics configuration in the HTTP Bridge container:
+
+1. `STRIMZI_METRICS`: This will contain the `metricsConfig` types to enable (one of `jmxPrometheusExporter` and `strimziMetricsReporter`).
+2. `KAFKA_BRIDGE_METRICS_JMX_CONFIG`: Used with `STRIMZI_METRICS=jmxPrometheusExporter` to pass the configuration file path.
+3. `KAFKA_BRIDGE_METRICS_SMR_CONFIG`: Used with `STRIMZI_METRICS=strimziMetricsReporter` to pass the plugin configuration.
+
+The `enableMetrics` property will be deprecated and removed in a future release.
+In case they are both set, `metricsConfig` will take precedence over `enableMetrics`.
+
+All Strimzi Metrics Reporter configurations will be locked down with the exception of `allowList`.
+Unlike the other major components, the metrics endpoint port will be set to 8080, which is the one already used to expose the REST API.
```

Comment:
the metrics endpoint should be disabled not set to 8080 otherwise it will conflict with the bridge one, or?

Reply:
Yes, I just expressed this poorly. Fixing.